Can't close a patient without  first editing something

## What went wrong? 😲
<!-- Provide a clear and concise description of what the bug is. Screenshots are helpful! --> 

![image](https://github.com/msupply-foundation/open-msupply/assets/6196269/1e854473-a1aa-4040-b1b1-9f0290f2c2d0)

The Cancel button is disabled until you edit something => you can't just close it. I know that the browser back button also works, but I don't think we should be relying on that in a "proper" application...

## Expected behaviour 🤔

Should be able to close the patient without editing it first!

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. Go to Dispensary -> Patients
2. Click on a patient record

## Your environment 🌱
<!-- e.g. 1.2.3 -->
 - Version: 1.5.0
 - Platform:
   - browser (Opera on Windows 11)
   - server (Windows)

I am trying to run `yarn pre-commit-lint` and I run into the error `Error: Cannot find module '@nx/nx-darwin-arm64'`.
When in try to run `yarn add -W @nx/nx-darwin-arm64` I fall into this error `@nx/nx-darwin-arm64@17.1.3: The CPU architecture "x64" is incompatible with this module.`
Is there some way i can fix this issue, because i can't commit my changes?

--- a/client/packages/system/src/Patient/PatientView/Footer.tsx
+++ b/client/packages/system/src/Patient/PatientView/Footer.tsx
@@ -23,6 +23,7 @@ interface FooterProps {
   validationError?: string | boolean;
   inputData?: FormInputData;
   showSaveConfirmation: () => void;
+  showCancelConfirmation: () => void;
 }
 
 export const Footer: FC<FooterProps> = ({
@@ -32,6 +33,7 @@ export const Footer: FC<FooterProps> = ({
   validationError,
   inputData,
   showSaveConfirmation,
+  showCancelConfirmation,
 }) => {
   const t = useTranslation();
   const { Modal, showDialog, hideDialog } = useDialog();
@@ -64,9 +66,13 @@ export const Footer: FC<FooterProps> = ({
           >
             <DialogButton
               variant="cancel"
-              disabled={!isDirty || isSaving}
+              disabled={isSaving}
               onClick={() => {
-                navigate(-1);
+                if (isDirty) {
+                  showCancelConfirmation();
+                } else {
+                  navigate(-1);
+                }
               }}
             />
             <LoadingButton

--- a/client/packages/system/src/Patient/PatientView/PatientView.tsx
+++ b/client/packages/system/src/Patient/PatientView/PatientView.tsx
@@ -13,6 +13,7 @@ import {
   UpdatePatientInput,
   BasicSpinner,
   DocumentRegistryCategoryNode,
+  useNavigate,
 } from '@openmsupply-client/common';
 import { usePatient } from '../api';
 import { AppBarButtons } from './AppBarButtons';
@@ -106,6 +107,7 @@ const PatientDetailView = ({
       },
     });
   const isLoading = isCurrentPatientLoading || isPatientRegistryLoading;
+  const navigate = useNavigate();
 
   const patientRegistry = patientRegistries?.nodes[0];
   const isCreatingPatient = !!createNewPatient;
@@ -228,6 +230,14 @@ const PatientDetailView = ({
     title: t('heading.are-you-sure'),
   });
 
+  const showCancelConfirmation = useConfirmationModal({
+    onConfirm: () => {
+      navigate(-1);
+    },
+    message: t('messages.confirm-cancel-generic'),
+    title: t('heading.are-you-sure'),
+  });
+
   if (isLoading) return <BasicSpinner />;
 
   return (
@@ -240,6 +250,7 @@ const PatientDetailView = ({
         validationError={validationError}
         inputData={inputData}
         showSaveConfirmation={showSaveConfirmation}
+        showCancelConfirmation={showCancelConfirmation}
       />
     </Box>
   );

